xapian-core/api/queryinternal.h

```diff
@@ -72,6 +76,10 @@ class QueryTerm : public Query::Internal {
 
     void serialise(std::string & result) const;
 
+    void set_weight(const Xapian::Weight& wt_) noexcept {
+	weight.reset(wt_.clone());
+    }
```

Comment:
`Query` objects are meant to act as if they are immutable, so a setter method like this doesn't really work properly.

I say "as if" - mostly they actually are immutable, but methods like `&=` are optimised for a `Query` with a ref count of 1 on its internals to modify in place so `&=` can be used to assemble a query in a loop and the loop can be `O(n)` overall.

I think this would need to be done via something like an `OP_WEIGHTING_SCHEME` operator which takes a subquery and a `Xapian::Weight` object to use for that subquery.

Reply:
Sure, that sounds better. So, we want to have a new Query operator
Query(const std::string & term,
           Xapian::Weight* weight,
           Xapian::termcount wqf = 1,
           Xapian::termpos pos = 0)

which internally instantiates a QueryTerm object with Weight object set via the constructor.